multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/util/StagingApplicationAttributeUpdater.java

```diff
@@ -44,7 +44,16 @@ private boolean hasStagingChanged(Staging staging, CloudApplication existingApp,
             processContext.setVariable(Variables.APP_NEEDS_RESTAGE, true);
             return true;
         }
-        return !healthCheck.equals(existingHealthCheck) || areAppFeaturesChanged(staging.getAppFeatures(), existingAppFeatures);
+        return !healthCheck.equals(existingHealthCheck) || areAppFeaturesChanged(staging.getAppFeatures(), existingAppFeatures)
+            || !hasReadinessHealthCheckChanged(staging);
+    }
+
+    private boolean hasReadinessHealthCheckChanged(Staging newStaging) {
```

Comment:
Move this method so it is below areAppFeaturesChanged
Rename it because hasChanged is true when the readiness health check is actually not changed, which is counterintuitive